ste/JobsAdmin.go

```diff
@@ -637,6 +637,8 @@ func (ja *jobsAdmin) ResurrectJob(jobId common.JobID, sourceSAS string, destinat
 		mmf := planFile.Map()
 		jm := ja.JobMgrEnsureExists(jobID, mmf.Plan().LogLevel, "")
 		jm.AddJobPart(partNum, planFile, sourceSAS, destinationSAS, false)
+		// Note that unmapping this mmf is perfectly fine. It just gets re-mapped in the job part managers.
+		mmf.Unmap()
```

Comment:
This seems unexpected, to close here so immediately and so automatically.  I find it hard to reason about the correctness of this in general.  Is there any way to replace this Unmap and the one added to RessurectJobParts, with explict close calls like you added to the ListJobs function?  E.g. closing from something in the call chain to this routine here, rather than unmapping right here?

Reply:
By "in general" I mean, in cases such as copy and sync and resume.

Reply:
This MMF never gets linked to a job part manager, and for all intents and purposes, never actually gets used again.

There is no explicit close call to be made, unless I'm misunderstanding your question?

Reply:
If anything, the file handle is otherwise leaked here.

Reply:
Do you mean "This MMF never gets linked to a job part manager when we run jobs clean"; or do you mean "This MMF never gets linked to a job part manager, and that's true no matter how this routine is called and used".

Reply:
I mean this MMF is literally never used again, no matter how it's called. I tested copy and resume before I sent in this PR because I had similar concerns.

Reply:
And is that also true (and tested) for the similar change that's been made to ResurrectJobParts?

Reply:
Yup. ResurrectJobParts is only used by ListJobs at the moment, and even then, it's _effectively_ just repeatedly calling ResurrectJob.

Reply:
So why can't the unmapping be moved into ListJobs?  That way, the ressurect* methods would be _safe_ to call in future from other places. Right now, they're not really safe to reuse, because (with this PR) they are making the assumption that nothing will actually use the plan file. Assuming "my caller will never actually use my data" seems like a likely way to surprise some future maintainer of the code.  
Thoughts on moving the unmapping to ListBlobs?

Reply:
The thing is that these mmfs literally don't get returned. They never get used again, because they can't be used again. Closing the mmfs here isn't unsafe, not closing them is just leaking file handles.

Reply:
The only thing this mmf is created for is to get the log level.

Reply:
It's _absolutely_ obtuse, and it would be great if AddJobParts just took in the MMF instead of repeating mapping in the file.

Reply:
Would you prefer if I set it up that way, so that the close call occurring later in the list call is the only place we need to close the mmf?

Reply:
By "that way", do you mean "leave the Resurrect* methods back the way they were before this PR, and instead close their MMFs in the List call"?  If so, yes, I think I'd find that easier to review and approve. 

(I'll find it hard to approve if the methods named Resurrect* actually give back objects that are partially "dead" - as is the case with the current draft of this PR.  I take your point that the existing logic and naming, prior to this PR, is not ideal.  But I'm not sure I can bring myself to get on board with Resurrect* returning a "dead"-ish object - since that's opposite of what 'resurrect" means).

Reply:
I think labeling these returned objects as "partially dead" is an incorrect assumption.

These MMFs aren't tracked. They aren't returned. They aren't reused anywhere. They are, for all intents and purposes, leaked memory and handles. Killing them doesn't kill anything about the underlying job managers, job part managers, or job part transfer managers. They're left _entirely_ in tact.

My suggestion is to make it so that calling AddJobPart allows you to take in an existing MMF, which would allow us to avoid that leakage.